File: client/python/example/quick/testCard.py
import grpc
import card_pb2

from example.card.card import CardSvc

NUM_OF_NEW_BLACKLIST = 2
FIRST_BLACKLISTED_CARD_ID = 100000
ISSUE_COUNT = 3

def makeCardInfos():
  cardInfos = []

  for i in range(0, NUM_OF_NEW_BLACKLIST):
    buf = str(FIRST_BLACKLISTED_CARD_ID + i).encode()
    cardInfo = card_pb2.BlacklistItem(cardID=buf, issueCount=ISSUE_COUNT)
    cardInfos.append(cardInfo)

  return cardInfos


def testCard(cardSvc, deviceID):
  try:
    print('>>> Scan a card...', flush=True)

    cardData = cardSvc.scan(deviceID)
    print(f'Card data: {cardData}', flush=True)

    blacklist = cardSvc.getBlacklist(deviceID)
    print(f'Initial blacklist: {blacklist}', flush=True)

    cardInfos = makeCardInfos()
    cardSvc.addBlacklist(deviceID, cardInfos)

    blacklist = cardSvc.getBlacklist(deviceID)
    print(f'Blacklist after adding cards: {blacklist}', flush=True)

    cardSvc.deleteBlacklist(deviceID, cardInfos)

    blacklist = cardSvc.getBlacklist(deviceID)
    print(f'Blacklist after deleting cards: {blacklist}', flush=True)

  except grpc.RpcError as e:
    print(f'Cannot complete the card test: {e}')

File: client/node/example/quick/testCard.js
const card = require('../card');

const NUM_OF_NEW_BLACKLIST = 2;
const	FIRST_BLACKLISTED_CARD_ID = 100000;
const ISSUE_COUNT = 3;

function makeCardInfos() {
  var cardInfos = [];

  for(i = 0; i < NUM_OF_NEW_BLACKLIST; i++) {
    let item = new card.cardMessage.BlacklistItem();
    item.setCardid(Buffer.from(`${FIRST_BLACKLISTED_CARD_ID + i}`, 'utf-8'));
    item.setIssuecount(ISSUE_COUNT);

    cardInfos.push(item);
  }

  return cardInfos;
}

function test(devID) {
	console.log('>>> Scan a card...');

  return card.scan(devID)
  .then((cardData) => {
    console.log('Card: ', cardData);
    return card.getBlacklist(devID);
  })
  .then((blacklist) => {
    console.log('Initial blacklist: ', blacklist);

    return card.addBlacklist(devID, makeCardInfos());
  })
  .then(() => {
    return card.getBlacklist(devID);
  })
  .then((blacklist) => {
    console.log('Blacklist after adding new items: ', blacklist);

    return card.deleteBlacklist(devID, makeCardInfos());
  })
  .then(() => {
    return card.getBlacklist(devID);
  })
  .then((blacklist) => {
    console.log('Blacklist after deleting new items: ', blacklist);

    return card.getConfig(devID);
  })
  .then((config) => {
    console.log('Card config: ', config);
  })
  .catch((err) => {
    console.error('Cannot finish the card test: ', err);
    throw err;
  });
}

module.exports.test = test;